bag_verifier/src/test/scala/uk/ac/wellcome/platform/archive/bagverifier/verify/steps/VerifyFilenamesTest.scala

```diff
@@ -0,0 +1,114 @@
+package uk.ac.wellcome.platform.archive.bagverifier.verify.steps
+
+import org.scalatest.EitherValues
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+import uk.ac.wellcome.platform.archive.common.bagit.models.{
+  BagPath,
+  PayloadManifest,
+  TagManifest
+}
+import uk.ac.wellcome.platform.archive.common.verify.{ChecksumValue, MD5}
+
+class VerifyFilenamesTest extends AnyFunSpec with Matchers with EitherValues {
+  val verifier: VerifyFilenames =
+    new VerifyFilenames {}
+
+  describe("verifyAllowedFilenames") {
+    it("allows legal filenames") {
+      verifier.verifyAllowedFilenames(Seq("cat.jpg", "dog.png", "fish.gif")) shouldBe Right(
+        ()
+      )
+    }
+
+    it("flags a file with a trailing dot") {
+      val err = verifier.verifyAllowedFilenames(Seq("bad.jpg.")).left.value
+      err.e.getMessage shouldBe "Filenames cannot end with a .: bad.jpg."
+      err.userMessage shouldBe Some("Filenames cannot end with a .: bad.jpg.")
+    }
+
+    it("flags multiple files with a trailing dot") {
+      val err = verifier
+        .verifyAllowedFilenames(Seq("bad.jpg.", "alsobad.png.", "good.tif"))
+        .left
+        .value
+      err.e.getMessage shouldBe "Filenames cannot end with a .: bad.jpg., alsobad.png."
+      err.userMessage shouldBe Some(
+        "Filenames cannot end with a .: bad.jpg., alsobad.png."
+      )
+    }
+  }
+
+  describe("verifyPayloadFilenames") {
+    it("allows filenames that start with data/") {
+      val manifest = PayloadManifest(
+        checksumAlgorithm = MD5,
+        entries = Map(
+          BagPath("data/README.txt") -> ChecksumValue("123"),
```

Comment:
Do we really need this check?

Reply:
The BagIt spec requires this, so I think so.

It also enforces an assumption that's only been implicit, which is that "BagIt metadata" and "interesting files" live in two separate directories.